.github/workflows/update_chart_dependencies.yaml

```
@@ -19,71 +19,64 @@ on:
-          LATEST_VER=$(helm search repo ${{ matrix.repo }} --versions | awk 'NR==2{print $2}')
-          echo "LATEST_VER=$LATEST_VER" >> $GITHUB_OUTPUT
-
-          # Retrieve the current version from chart.yaml
-          DEP_PATH=$(yq eval ".dependencies[] | select(.name == \"${{ matrix.repo }}\") | .version" $CHART_YAML)
-
-          echo "Current version of ${{ matrix.repo }} is $DEP_PATH, latest is $LATEST_VER"
 
-          if [ "$LATEST_VER" == "$DEP_PATH" ]; then
-            echo We are already up to date. Nothing else to do.
-          else
-            echo 'Looks like we need to update...'
-            echo Updating to new version in chart.yaml
-            echo "NEED_UPDATE=1" >> $GITHUB_OUTPUT
+      - name: Check for Version Updates
+        id: check_for_update
+        run: |
+          echo "Checking chart dependency version for ${{ matrix.name }}"
 
-            # Update the version in chart.yaml using yq and jq
-            DEP_LINE=$(yq eval ".dependencies | keys | map(tonumber) | map(select(. != null)) | map(select(. < 10000)) | map(. + 1)" $CHART_YAML | jq ".[] | select(.[\"name\"] == \"${{ matrix.repo }}\")")
-            sed -i "${DEP_LINE}s/$DEP_PATH/$LATEST_VER/" $CHART_YAML
+          make update-chart-dep CHART_PATH=${{ matrix.yaml_file_path }} SUBCHART_NAME='${{ matrix.dependency_name }}' DEBUG_MODE=$DEBUG_MODE || exit 1
 
-            if [ ${{ matrix.repo }} == "opentelemetry-operator" ]; then
-              echo "Update Splunk Operator Instrumentation Images if Needed"
-              ./ci_scripts/update-images-operator-otel.sh ${{ matrix.language }} $DEBUG_ARG || exit 1
-            fi
+      - name: Open PR for Version Update
+        id: open_pr
+        if: ${{ steps.check_for_update.outputs.NEED_UPDATE == 1 }}
+        uses: peter-evans/create-pull-request@v5
+        with:
+          commit-message: Update ${{ matrix.name }} chart dependency version
+          title: Bump ${{ matrix.name }} from ${{ steps.check_for_update.outputs.CURRENT_VER }} to ${{ steps.check_for_update.outputs.LATEST_VER }} in ${{ matrix.yaml_file_path }}
+          body: Use the latest version of ${{ matrix.name }}
+          branch: "update-${{ matrix.name }}" # Same branch name for all PRs
+          base: main
+          delete-branch: true
+          modify-outputs: false
 
-            echo Updating rendered examples
-            make render
+      - name: Apply Version Update and Generate Changelog
+        if: ${{ steps.check_for_update.outputs.NEED_UPDATE == 1 }}
+        run: |
+          # Apply the version update, update the rendered examples with the version update, and create a changelog entry
+          # We run `make update-chart-dep` again here because the open_pr peter-evans/create-pull-request step before clears out the update changes locally
+          make update-chart-dep CHART_PATH=${{ matrix.yaml_file_path }} SUBCHART_NAME='${{ matrix.dependency_name }}' DEBUG_MODE=$DEBUG_MODE || exit 1
```

```suggestion
          make update-chart-dep CHART_PATH=${{ matrix.yaml_file_path }} SUBCHART_NAME='${{ matrix.dependency_name }}' DEBUG_MODE=$DEBUG_MODE
```
Github by [default runs shell with `set -e`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell), I think we can drop the explicit `|| exit 1`.

---

Good point, removed the extra checks from this and a similar workflow